src/python/pants/engine/selectors.py

```diff
@@ -121,13 +120,26 @@ def __repr__(self):
       field_name_portion = ', {}'.format(repr(self.field))
     else:
       field_name_portion = ''
-    return '{}({}, {}{}{}{})'.format(type(self).__name__,
```

Comment:
I don't think this overridden `repr` is still used anywhere... may be able to remove it.